ui/SpinInfoBar.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { useEffect, useState } from 'react';
+import { useDark } from './lib/useDark';
+
+export const SpinInfoBar = () => {
+  const [spinStack, setSpinStack] = useState<string[]>([]);
+  const dark = useDark();
+
+  useEffect(() => {
+    console.log('spin: setting up...');
+    let available = false;
+    const spin = new EventSource('/~_~/spin', { withCredentials: true });
+
+    spin.onopen = () => {
+      console.log('spin: opened stream');
+      available = true;
+    };
+
+    spin.onmessage = (e) => {
+      console.log('spin message:', e.data);
+      // The data is expected to be a path like /stack3/stack2/stack1/stack0
+      // We need to parse it into an array of stack names
+      if (e.data && typeof e.data === 'string') {
+        // Remove leading slash if present and split by slash
+        const stackPath = e.data.startsWith('/') ? e.data.substring(1) : e.data;
+        const stackArray = stackPath.split('/').filter(Boolean);
+        setSpinStack(stackArray);
```

Comment:
Isn't the leading slash always present? If we're gonna check, we should just drop/ignore malformed messages where the expected leading slash is missing.

Also unclear what the `.filter()` call does here, feels like you could just drop it, no? Even if it filters empty ones, we'll want to retain & render those, no?

```suggestion
        const stackPath = e.data.substring(1);
        const stackArray = stackPath.split('/');
        setSpinStack(stackArray);
```

Reply:
Fixed